### PR TITLE
Documentation updates around environments.

### PR DESCRIPTION
**Pull request by adunkman**
In this PR:

- Organizes [`docs/README.md`](https://github.com/ustaxcourt/ef-cms/blob/setup-documentation-updates/docs/README.md) into sections. 
- Organizes [`docs/SETUP.md`](https://github.com/ustaxcourt/ef-cms/blob/setup-documentation-updates/docs/environments/setup.md) into sections to help clarify what needs to be done when creating new AWS environments (registering for new accounts isn’t needed, for example) and moves to new `environments` folder.
- Clearly documents the manual deployment steps needed in [`docs/environments/release.md`](https://github.com/ustaxcourt/ef-cms/blob/setup-documentation-updates/docs/environments/release.md). This replaces and is an iteration of `docs/branch-management.md`, where the environment-specific info is in the [Court-specific environment documentation in the Wiki](https://github.com/ustaxcourt/ef-cms/wiki/AWS-environment-configuration). 
- Adds [`docs/environments/teardown.md`](https://github.com/ustaxcourt/ef-cms/blob/setup-documentation-updates/docs/environments/teardown.md) to describe how to destroy an environment.

### CODE REVIEW

**Comment by adunkman** [comment]
@ericsorenson @julialeague [documented the teardown script](https://github.com/ustaxcourt/ef-cms/blob/setup-documentation-updates/docs/environments/teardown.md) — two items for y’all!

1. Would you confirm that there are no other steps that you have performed when destroying environments (no additional scripts, manual checks, or verifications)? I want to make sure we’re documenting the process in reality, not as idealized.

2.  Would either of you be able to cross-reference issues or pull requests for the two known issues? I’m sure they’re tracked somewhere, but I wasn’t able to find them.

**Comment by adunkman** [comment]
Improving the automation on setup pulled into https://github.com/ustaxcourt/ef-cms/issues/286.

**Comment by adunkman** [comment]
Asked about missing environment configuration via [Slack](https://ustaxcourt.slack.com/archives/CD4PNKEPK/p1594136976347700).

**Comment by adunkman** [comment]
> Environment variable configuration should be reconciled against current variables in the Court and Flexion’s environments to discover what else is missing.

Completed, and removed stale configuration. No additional configuration beyond Honeybadger is needed.

**Comment by adunkman** [comment]
Ready for review!

File: docs/README.md [hunk 1]
```diff
@@ -1,44 +1,37 @@
...
+| API documentation | The API is documented via Swagger, but right now it’s only available when the site is running in Docker. That can be reviewed at [`http://localhost:3000/api/swagger`](http://localhost:3000/api/swagger).
+| HTML style guide | There is an HTML style guide, but right now it’s only available when the site is running in Docker. That can be found at [`http://localhost:1234/style-guide`](http://localhost:1234/style-guide).
+|  Visual style guide | A [visual style guide](style-guide.pdf) — covering typography, colors, icons, buttons, form elements, etc. — is maintained as a PDF.
+| Accessibility testing | Per the requirements of the code review, all work must meet [Section 508 standards](https://www.section508.gov/), per [WCAG 2.0 AA](https://www.w3.org/TR/WCAG20/). The process by which that’s assured by developers is documented in [`ACCESSIBILITY.md`](ACCESSIBILITY.md).
+| Developer Check Lists | Various check lists that can help you out when developing on this project and doing things such as adding a new endpoint or creating a new stack. [`CHECKLISTS.md`](CHECKLISTS.md).
```
Comment (julialeague) [inline docs/README.md:27]: This doc is now outdated after removing Serverless, I think. I'd like to follow up with Flexion to update it.

**Pull request merged**